sdk/plugin/plugin.go

```diff
@@ -78,24 +92,30 @@ func NewPluginClient(ctx context.Context, sys pluginutil.RunnerUtil, pluginRunne
 		// and version 4.
 		3: {
 			"backend": &GRPCBackendPlugin{
-				MetadataMode: isMetadataMode,
+				MetadataMode: config.IsMetadataMode,
 			},
 		},
 		4: {
 			"backend": &GRPCBackendPlugin{
-				MetadataMode: isMetadataMode,
+				MetadataMode: config.IsMetadataMode,
+			},
+		},
+		5: {
+			"backend": &GRPCBackendPlugin{
+				MetadataMode:      false,
+				AutoMTLSSupported: true,
 			},
 		},
 	}
 
-	namedLogger := logger.Named(pluginRunner.Name)
-
 	var client *plugin.Client
 	var err error
-	if isMetadataMode {
-		client, err = pluginRunner.RunMetadataMode(ctx, sys, pluginSet, handshakeConfig, []string{}, namedLogger)
+	if config.IsMetadataMode {
+		client, err = pluginRunner.RunMetadataMode(ctx, config.Wrapper, pluginSet, handshakeConfig, []string{}, config.Logger)
+	} else if config.AutoMTLS {
```

Comment:
I wonder if the check for `config.AutoMTLS` should come first to prioritize running plugins with autoMTLS if that's intended.